kore/test/Test/ConsistentKore.hs

```diff
@@ -818,14 +794,18 @@ symbolGenerator
       where
         applicationGenerator termGenerator sort = do
             when (sort /= applicationSortsResult) (error "Sort mismatch.")
-            let request =
-                    if BuiltinMap.isSymbolElement symbol
-                        || BuiltinSet.isSymbolElement symbol
-                        then requestConcrete . requestConstructorLike
-                        else -- TODO (virgil): also allow constructor-like stuff
-                        -- with variables.
-                            id
-            maybeTerms <- request $ mapM termGenerator applicationSortsOperands
+            let restrict
+                    | BuiltinMap.isSymbolElement symbol =
+                        requestConcrete . requestConstructorLike
+                    | BuiltinSet.isSymbolElement symbol =
+                        requestConcrete . requestConstructorLike
+                    | isFunction =
+                        withoutConnectives
+                    | otherwise = id
+                -- TODO (virgil): also allow constructor-like
+                -- stuff with variables.
+                isFunction = Attribute.Symbol.isFunction symbolAttributes
```

Comment:
why not inline above?

Reply:
Good point... the separation was from before when I hadn't found that `isFunction` is ready-made. Will fix it.